### ISSUE TITLE
exec-path ordering is inconsistent with .spacemacs.env

### ISSUE THREAD

**Issue by leezu** [syl20bnr/spacemacs#12403]
#### Description :octocat:
For example, when putting
/path/to/my/home/.pyenv/bin/:/path/to/my/home/.pyenv/shims/ at the front of the
PATH in .spacemacs.env, they appear in the middle of the exec-path variable.
/usr/bin appears before
/path/to/my/home/.pyenv/bin/:/path/to/my/home/.pyenv/shims/ in exec-path, thus
takes precedence.

#### Reproduction guide :beetle:
- Configure PATH in .spacemacs.env
- Start Emacs

*Observed behaviour:* :eyes: :broken_heart:
- exec-path ordering is inconsistent with .spacemacs.env

*Expected behaviour:* :heart: :smile:
- exec-path ordering is the same as in .spacemacs.env

#### System Info :computer:
- OS: gnu/linux
- Emacs: 26.2
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. fc3677131)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm
 (auto-completion :variables auto-completion-enable-sort-by-usage t auto-completion-enable-help-tooltip t)
 (spell-checking :variables spell-checking-enable-by-default nil enable-flyspell-auto-completion t)
 vinegar my-ui multiple-cursors neotree bibtex
 (org :variables org-projectile-file "~/org/projectile.org" org-want-todo-bindings t org-agenda-span 1 org-agenda-start-on-weekday nil)
 (clip2org :variables clip2org-clippings-file "~/Dropbox/Books/My Clippings.txt")
 my-org
 (deft :variables deft-directory "~/org/deft")
 (my-mail :variables notmuch-delete-tags
          '("+deleted" "-inbox" "-autoinbox")
          notmuch-spam-tags
          '("+spam" "-inbox" "-autoinbox")
          notmuch-kill-tags
          '("+killed" "-inbox" "-autoinbox"))
 finance dash sphinx cscope emacs-lisp package-lint git github
 (latex :variables latex-enable-folding t latex-build-command "LatexMk")
 restructuredtext markdown pandoc syntax-checking
 (ess :variables ess-disable-underscore-assign t)
 (python :variables python-enable-yapf-format-on-save nil)
 java go yaml lua semantic
 (c-c++ :variables c-c++-enable-clang-support t c-c++-enable-rtags-support t c-c++-default-mode-for-headers 'c++-mode)
 cmake octave
 (shell :variables shell-default-term-shell "/bin/zsh" shell-default-shell 'ansi-term)
 shell-scripts nixos csv systemd
 (html :variables web-fmt-tool 'prettier)
 (javascript :variables javascript-fmt-tool 'prettier js2-strict-missing-semi-warning nil js2-missing-semi-one-line-override nil)
 (json :variables json-fmt-tool 'prettier)
 prettier react
 (typescript :variables typescript-fmt-tool 'prettier)
 restclient spotify)
```
- System configuration features: XPM JPEG TIFF GIF PNG SOUND GSETTINGS GLIB NOTIFY GNUTLS FREETYPE XFT ZLIB TOOLKIT_SCROLL_BARS GTK2 X11 XDBE XIM THREADS

**Comment by Compro-Prasad** [comment]
There is a new PR about this. Can you try [it](https://github.com/syl20bnr/spacemacs/pull/12417) out? Make sure you backup and clean `.spacemacs.env` before trying it out.

**Comment by mgourd** [comment]
@Compro-Prasad I tried, it doesn't fix this issue.
If I leave `dotspacemacs/user-env` as default, .spacemacs.env is not loaded.
If I manually `(load-env-vars spacemacs-env-vars-file)`, `(getenv "PATH")` return the expected results, but in exec-path "/usr/bin" (in the case of macOS, "/usr/bin:/bin:/usr/sbin:/sbin:/Applications/Emacs.app/Contents/MacOS/bin-x86_64-10_14:/Applications/Emacs.app/Contents/MacOS/libexec-x86_64-10_14") is appended in front of PATH.

**Comment by pedrobellini** [comment]
I have similar issue... when launching emacs through a bat file (where I previously set the PATH variable, among others) seems to now mess up the environment once spacemacs launches, not giving priority anymore to spacemacs.env file . It was working properly before, so I'm guessing it's related to a recent update.
Same behaviour after a clean spacemacs.env file was generated.

**Comment by duianto** [comment]
>It was working properly before, so I'm guessing it's related to a recent update.

This environment related PR that was merged recently:
[core] Environment variable caching fixes #12417

**Comment by jaminsore** [comment]
Since my config is shared between my home and work computers (ubuntu and mac os, respcectively) My workaround was to:
1. add these lines to my `spacemacs/user-init()`
```elisp
  (defvar path-file
    "path to elisp file that explicitly defines and sets both
    `exec-path' and `PATH'")

  (setq path-file "~/.spacemacs.pth")

  (when path-file
    (load-file path-file))
```
2. remove the `PATH` entry from my `.spacemacs.env` file
3. and add these lines to my "~/.spacemacs.pth`
```elisp
(let ((path "the:path:entries:I:want:formerly:located:in:.spacemacs.env"))
  (setenv "PATH" path)
  (setq exec-path (split-string path ":")))
```

Now my config is both immune to whatever `exec-path` fiddling occurs in the future and the same config can be used on my work and home computers.

**Comment by saiberz** [comment]
I had the same error but I see it has been solved in #13858
